metals/src/main/scala/scala/meta/internal/metals/ServerCommands.scala

```diff
@@ -231,6 +231,23 @@ object ServerCommands {
        |""".stripMargin,
   )
 
+  val DiscoverJvmRunCommand = new ParametrizedCommand[DebugDiscoveryParams](
+    "discover-jvm-run-command",
+    "Discover main classes to run.",
+    """|Gets the DebugSession object that also contains a command to run in shell based 
+       |on JVM environment including classpath, jvmOptions and environment parameters.
+       |""".stripMargin,
+    """|DebugUnresolvedTestClassParams object
+       |Example:
+       |```json
+       |{
+       |   testClass: "com.foo.FooSuite",
+       |   buildTarget: "foo"
+       |}
+       |```
+       |""".stripMargin,
+  )
```

Comment:
- use named arguments - it increases code readability in significant way
- `DebugUnresolvedTestClassParams` isn't an argument for this command, it is `DebugDiscoveryParams` looking at types
- can you add an example what does response look like? 
- `Gets the DebugSession object` it's a bit weird that we mix debugging and running :/ Have to look at vscode to see whole picture though
- `Gets the DebugSession object that also contains a command to run in shell` - I'm afraid that this is a lie. It _may return such extended DebugSession_

Reply:
Let's do that in a follow up.